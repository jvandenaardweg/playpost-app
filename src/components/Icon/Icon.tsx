/**
 * Multiple ways to import:
 * 1- Importing a specific font as Icon
 * -- import Icon from './lib/Icons/FontAwesome'
 *
 * 2- Importing a specific font as its name
 * -- import {FontAwesome} from './lib/Icons'
 *
 * 3- Importing all fonts in an Icon object
 * -- import * as Icon from './lib/Icons'
 */
import React from 'react';
// import MaterialCommunityIconsI from 'react-native-vector-icons/MaterialCommunityIcons';
// import SimpleLineIconsI from 'react-native-vector-icons/SimpleLineIcons';
// import MaterialIconsI from 'react-native-vector-icons/MaterialIcons';
import FontAwesomeI from 'react-native-vector-icons/FontAwesome';
import FontAwesome5I, { FontAwesome5IconProps } from 'react-native-vector-icons/FontAwesome5';
// import FoundationI from 'react-native-vector-icons/Foundation';
// import EvilIconsI from 'react-native-vector-icons/EvilIcons';
// import OcticonsI from 'react-native-vector-icons/Octicons';
// import IoniconsI from 'react-native-vector-icons/Ionicons';
import FeatherI from 'react-native-vector-icons/Feather';
// import EntypoI from 'react-native-vector-icons/Entypo';
// import ZocialI from 'react-native-vector-icons/Zocial';
import { IconProps } from 'react-native-vector-icons/Icon';

// export const MaterialCommunityIcons = (props: IconProps) => <MaterialCommunityIconsI {...props} />;)
// export const SimpleLineIcons = React.memo((props: IconProps) => <SimpleLineIconsI {...props} />);
// export const MaterialIcons = React.memo((props: IconProps) => <MaterialIconsI {...props} />);
export const FontAwesome = React.memo((props: IconProps) => <FontAwesomeI {...props} />);
export const FontAwesome5 = React.memo((props: FontAwesome5IconProps) => <FontAwesome5I {...props} />);
// export const Foundation = React.memo((props: IconProps) => <FoundationI {...props} />);
// export const EvilIcons = React.memo((props: IconProps) => <EvilIconsI {...props} />);
// export const Ionicons = React.memo((props: IconProps) => <IoniconsI {...props} />);
// export const Octicons = React.memo((props: IconProps) => <OcticonsI {...props} />);
export const Feather = React.memo((props: IconProps) => <FeatherI {...props} />);
// export const Entypo = React.memo((props: IconProps) => <EntypoI {...props} />);
// export const Zocial = React.memo((props: IconProps) => <ZocialI {...props} />);
