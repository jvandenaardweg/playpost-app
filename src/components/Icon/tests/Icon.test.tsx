import React from 'react';
import { render } from 'react-native-testing-library';

import * as Icon from '../Icon';

describe('Icon', () => {

  describe('rendering', () => {
    it('should render FontAwesome5 icon', () => {
      const icon = render(<Icon.FontAwesome5 name="heart" />);
      expect(icon.toJSON()).toMatchSnapshot();
    });

    it('should render FontAwesome icon', () => {
      const icon = render(<Icon.FontAwesome name="heart" />);
      expect(icon.toJSON()).toMatchSnapshot();
    });

    it('should render Feather icon', () => {
      const icon = render(<Icon.Feather name="heart" />);
      expect(icon.toJSON()).toMatchSnapshot();
    });
  });
});
