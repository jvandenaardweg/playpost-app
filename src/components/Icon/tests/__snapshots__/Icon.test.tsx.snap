// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon rendering should render Feather icon 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Feather",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Icon rendering should render FontAwesome icon 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Icon rendering should render FontAwesome5 icon 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome5Free-Regular",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {
        "fontWeight": "400",
      },
    ]
  }
>
  
</Text>
`;

exports[`Icon rendering should render correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
